apps/spruce/src/pages/waterfall/utils.test.ts

```diff
@@ -137,6 +139,276 @@ describe("groupBuildVariants", () => {
   });
 });
 
+describe("caching functions", () => {
+  describe("mergeVersions", () => {
```

Comment:
could we pull the `readField` definition up so we only have to write ts-expect-error and the function definition once at the top?
```
  // @ts-expect-error
  const readField: ReadFieldFunction = (field, obj) => obj[field];
```